src/Trace/Trace.php

```diff
@@ -0,0 +1,211 @@
+ * project and have many TraceSpans. See:
+ * https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#resource-trace
+ */
+class Trace
+{
+    use ValidateTrait;
+
+    /**
+     * @var ConnectionInterface Represents a connection to Stackdriver Trace.
+     */
+    private $connection;
+
+    /**
+     * The id of the project this trace belongs to.
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * The trace id for this trace. 128-bit numeric formatted as a 32-byte hex string
+     * @var string
+     */
+    private $traceId;
+
+    /**
+     * List of TraceSpans to report
+     * @var TraceSpan[]
+     */
+    private $spans = [];
+
+    /**
+     * Instantiate a new Trace instance.
+     *
+     * @param ConnectionInterface $connection The connection to Stackdriver Trace.
+     * @param string $projectId The id of the project this trace belongs to.
+     * @param string $traceId [optional] The id of the trace. If not provided, one will be generated
+     *        automatically for you.
+     * @param array $spans [optional] Array of TraceSpan constructor arguments. See
+     *        {@see Google\Cloud\Trace\TraceSpan::__construct()} for configuration details.
+     * }
+     */
+    public function __construct(ConnectionInterface $connection, $projectId, $traceId = null, $spans = null)
+    {
+        $this->connection = $connection;
+        $this->projectId = $projectId;
+        $this->traceId = $traceId ?: $this->generateTraceId();
+        if ($spans) {
+            $this->spans = array_map(function ($span) {
+                return new TraceSpan($span);
+            }, $spans);
+        }
+    }
+
+    /**
+     * Retrieves the trace's id.
+     *
+     * @return string
+     */
+    public function traceId()
+    {
+        return $this->traceId;
+    }
+
+    /**
+     * Retrieves the trace's projectId.
+     *
+     * @return string
+     */
+    public function projectId()
+    {
+        return $this->projectId;
+    }
+
+    /**
+     * Set the trace's projectId
+     *
+     * @param string $projectId
+     */
+    public function setProjectId($projectId)
+    {
+        $this->projectId = $projectId;
+    }
+
+    /**
+     * Returns a serializable array representing this trace. If no span data
+     * is cached, a network request will be made to retrieve it.
+     *
+     * @return array
+     */
+    public function info()
+    {
+        if (!$this->spans) {
+            $this->reload();
+        }
+
+        return [
+            'projectId' => $this->projectId,
+            'traceId' => $this->traceId,
+            'spans' => array_map(function ($span) {
+                return $span->info();
+            }, $this->spans)
+        ];
+    }
+
+    /**
+     * Triggers a network request to load a span's details.
+     *
+     * @see https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces/get
+     */
+    public function reload()
```

Comment:
Anywhere we do an API requests, we need to accept `array $options = []` as an argument, and pass it to the connection call:

```php
public function reload(array $options = [])
{
    $this->connection->getTrace([/**/] + $options);
}
```

The same applies to `Trace::info()` (pass $options into the `reload()` call.)